smarts/core/bubble_manager.py

```diff
@@ -563,10 +563,11 @@ def _prepare_sensors_for_agent_control(
         # Setup mission (also used for observations)
         # XXX: this is not quite right.  route may not be what the agent wants to take.
         route = sim.traffic_sim.vehicle_route(vehicle_id=vehicle.id)
-        mission = Mission(
-            start=Start(vehicle.position[:2], vehicle.heading),
-            goal=PositionalGoal.from_road(route[-1], sim.scenario.road_map),
-        )
+        if len(route) > 0:
+            goal = PositionalGoal.from_road(route[-1], sim.scenario.road_map)
+        else:
+            goal = EndlessGoal()
```

Comment:
I believe this was a bug with a 0 length route being considered a positional goal when there is no end edge.